cmd/traffic/cmd/agentinit/agent_init.go

```diff
@@ -30,7 +29,7 @@ func configureIptables(ctx context.Context, iptables *iptables.IPTables, loopbac
 	// However, if there's a service mesh we want to make sure we don't bypass the mesh, so the traffic will flow request -> mesh -> agent -> app
 	appPort := strconv.Itoa(cfg.AppPort)
 	agentPort := strconv.Itoa(cfg.AgentPort)
-	agentUID := strconv.FormatInt(install.AgentUID, 10)
+	agentUID := strconv.Itoa(os.Getuid())
```

Comment:
hmmmm the problem with this is that the application container will be running under the same UID as the sidecar, no? So i guess if the running application requests localhost it will actually get itself and not the intercept...

maybe that's okay but we should document it explicitly if so

Reply:
You mean if the app requests the `localhost:<intercepted port>`? Then yeah, I think it must be rerouted to itself, not the intercept. That behavior is consistent with what happens with the alternative approach, where we actually patch the service to use the port of our sidecar.